pkg/cache/cache_test.go

```diff
@@ -789,7 +799,97 @@ func CacheTest(createCacheFunc func(config *rest.Config, opts cache.Options) (ca
 					Eventually(out).Should(Receive(Equal(pod)))
 					close(done)
 				})
+				type selectorsTestCase struct {
+					fieldSelectors map[string]string
+					labelSelectors map[string]string
+					expectedPods   []string
+				}
+				DescribeTable("filter informers at list watch with selectors", func(tc selectorsTestCase) {
```

Comment:
DescribeTable is a perfect match to test this, is this important ?

Reply:
Its a tad unusual to read as the contents over which we loop come at the bottom instead of at the top like one would expect for a for loop

Reply:
Well this is how Ginkgo tests table works :-/ is really worth it to transform this into more usual go testing test table ?

Reply:
I guess I don't really understand why we need a ginkgo feature when we can just use a `for` loop?

Reply:
Well, since Ginkgo is already used, make sense to integrate with it, so we can use stuff like "-focus" or "-skip" with the different test table entries.

Reply:
@alvaroaleman are you ok with this ?

Reply:
Yeah, its fine. The "don't abuse client opts for this" remark is still open though.

Reply:
ack, resolving this.